File: readability/readability.c
// this programm will take a text from user and analyzes the text then returns the grade of the text given 
// importing libraries 
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("enter your text :");
    int letters = 0;
    int words = 1;
    int sentence = 0;

//using for loop the program will count words letters and sentences till the whole text is readed by program
    for (int i = 0; i < strlen(text); i++)
//counting letters 
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;   
        }
//counting words    
        else if (text[i] == ' ')
        {
            words++;
        }
//counting sentences    
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence++;
        }
    }     
// using the math formula given the program can calculate the grade of text a number between 0 and 16     
    float grade =  0.0588 * (100 * (float) letters / (float) words) - 0.296 * (100 * (float) sentence / (float) words) - 15.8;
// using if and else statement and the result of the math formula above,the program will tell the user the grad of their text.    
    if (grade >= 0 && grade < 16)
    {
        printf("Grade %i\n", (int) round(grade));
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else 
    {
        printf("Before Grade 1\n");
    }

}

File: readability py/readability.py
# including library
from cs50 import get_string
# making input
text = get_string("Text: ")
# creating variables
letters = words = sentences = 0
# starting to count letters,words and sentences
for char in text:
    if char.isalpha():
        letters += 1
    if char.isspace():
        words += 1
    if char in ['?', '.', '!']:
        sentences += 1

words += 1
# using this formula to grade texts
L = (letters * 100.0) / words
S = (sentences * 100.0) / words
result = int((0.0588 * L - 0.296 * S - 15.8) + 0.5)
if result < 1:
    print("Before Grade 1")
elif result >= 16:
    print("Grade 16+")
else:
    print(f"Grade {result}")